### PR TITLE
Assign `constructor` with `Object.defineProperty` for Safari 9.

### PR DESCRIPTION
**Pull request by bicknellr**

### CODE REVIEW

**Comment by bicknellr** [comment]
I haven't figured out why the tests are failing. The test pass in my local VMs but I can't get a sauce tunnel to stay open to be able to test in the sauce VMs.

**Comment by bicknellr** [comment]
@cdata, you might want to take another look. I switched the repo over to variants and removed the copied tests.

**Comment by bicknellr** [comment]
@cdata ping

**Comment by blasten** [comment]
friendly ping @cdata

File: test/test-fixture.html [hunk 1]
```diff
@@ -17,23 +17,42 @@
...
+      configurable: true,
+      enumerable: true,
+      writable: true,
+      value: XCustom,
+    });
+    XCustom.prototype.onDetached = function() {};
+    XCustom.prototype.detachedCallback =
+    XCustom.prototype.disconnectedCallback = function() {
+      this.onDetached();
+    };
```
Comment (cdata) [inline test/test-fixture.html]: I think this needs to be indented four additional spaces.

File: test/test-fixture-v1.html [hunk 2]
```diff
@@ -1,262 +0,0 @@
...
-subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
--->
-<html>
-<head>
-  <title>test-fixture</title>
-
-  <script>
-    // Save original HTMLElement constructor since customElements v1 polyfill
-    // overrides it https://github.com/webcomponents/webcomponentsjs/issues/623
-    var _HTMLElement = window.HTMLElement;
```
Comment (cdata) [inline test/test-fixture-v1.html]: Is it no longer necessary to do this?

File: bower.json [hunk 3]
```diff
@@ -9,9 +9,15 @@
     "testing"
   ],
   "devDependencies": {
-    "web-component-tester": "^4.0.0",
-    "webcomponentsjs": "webcomponents/webcomponentsjs#^0.7.0",
-    "webcomponentsjs-v1": "webcomponents/webcomponentsjs#v1"
+    "web-component-tester": "6.0.0-prerelease.5",
+    "webcomponentsjs": "webcomponents/webcomponentsjs#v1"
```
Comment (cdata) [inline bower.json]: Should these `devDependencies` be expressed as version ranges? (genuinely unsure)

File: test/test-fixture-v1.html (hunk 2 above)
Comment (bicknellr) [inline test/test-fixture-v1.html]: Yeah, `x instanceof HTMLElement` works with native elements now.

File: test/test-fixture.html (hunk 1 above)
Comment (bicknellr) [inline test/test-fixture.html]: fixed

File: bower.json (hunk 3 above)
Comment (bicknellr) [inline bower.json]: That sounds like the right thing to do? The [bower docs page about the dependencies field](https://github.com/bower/spec/blob/59c8f0e8f8444cbdd71a091919d28d0fa29d56fe/json.md#dependencies) doesn't seem to mention anything about being able to include version numbers when using a git url. :/ I'll try it out in a bit to see if it's there but undocumented.

Comment (bicknellr) [inline bower.json]: It seems like it works when I tried it locally so I updated the branch. I guess we'll see what happens when it tests the 1.x variant?

**Comment by bicknellr** [comment]
@justinfagnani, do all variants always install the latest applicable version of a dependency independently of other variants? Particularly, in 75b5627, is the default variant going to end up installing wct ^4.0.0 and webcomponentsjs ^0.7.0 since 1.x requires them and the default accepts them?

**Comment by bicknellr** [comment]
Talked offline, seems like the preferred way to be sure that a particular set of dependencies are being tested is to add an explicit variant.

**Comment by bicknellr** [comment]
@cdata ping

**Comment by bicknellr** [comment]
Talked offline, looks like the dependencies here don't work because branches aren't valid (or at least comparable) in semver ranges. So, we'll need a semver tag on webcomponentsjs to be able to express this dependency properly.

File: bower.json [hunk 4]
```diff
@@ -9,9 +9,22 @@
     "testing"
   ],
   "devDependencies": {
-    "web-component-tester": "^4.0.0",
-    "webcomponentsjs": "webcomponents/webcomponentsjs#^0.7.0",
-    "webcomponentsjs-v1": "webcomponents/webcomponentsjs#v1"
+    "web-component-tester": "^4.0.0 || 6.0.0-prerelease.5",
+    "webcomponentsjs": "webcomponents/webcomponentsjs#^0.7.0 || v1"
```
Comment (bicknellr) [inline bower.json]: `^0.7.0 || v1` isn't valid, we'll have to add a prerelease tag to webcomponentsjs. I think a versioning plan is being discussed at the moment.

Comment (cdata) [inline bower.json]: Let's pin it to the prerelease https://github.com/webcomponents/webcomponentsjs/releases/tag/v1.0.0-rc.1

File: bower.json [hunk 5]
```diff
@@ -9,9 +9,22 @@
...
+  },
+  "variants": {
+    "2.x": {
```
Comment (cdata) [inline bower.json]: Let's remove the 2.x variant.

File: bower.json [hunk 6]
```diff
@@ -9,9 +9,22 @@
     "testing"
   ],
   "devDependencies": {
-    "web-component-tester": "^4.0.0",
-    "webcomponentsjs": "webcomponents/webcomponentsjs#^0.7.0",
-    "webcomponentsjs-v1": "webcomponents/webcomponentsjs#v1"
+    "web-component-tester": "^4.0.0 || 6.0.0-prerelease.5",
```
Comment (cdata) [inline bower.json]: Do we need to pin this to the prerelease as well?

**Comment by bicknellr** [comment]
Looks like Firefox is having some trouble again, I'll restart the builds.

File: bower.json (hunk 5 above)
Comment (bicknellr) [inline bower.json]: removed

File: bower.json (hunk 6 above)
Comment (bicknellr) [inline bower.json]: Yeah, I think we're going to end up with the same problem here. Changed it to `^6.0.0-prerelease.5`.

**Pull request merged**